src/lib/clapper/clapper-harvest.c

```diff
@@ -0,0 +1,348 @@
+/* Clapper Playback Library
+ * Copyright (C) 2024 Rafał Dzięgiel <rafostar.github@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+/**
+ * ClapperHarvest:
+ *
+ * An object storing data from enhancers implementing [iface@Clapper.Extractable].
+ *
+ * Since: 0.8
+ */
+
+/*
+ * NOTE: We cannot simply expose GstMiniObjects for
+ * implementations to assemble TagList/Toc themselves, see:
+ * https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/2867
+ */
+
+#include "clapper-harvest-private.h"
+
+#define REQUEST_HEADERS_STRUCTURE_NAME "request-headers"
+#define UA_FIELD_NAME "User-Agent"
+#define DEFAULT_UA "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)"
```

Comment:
Should the default UA be googlebot-like, since this is not a crawler?

Reply:
Good point. Changed to Tor browser default. This is meant only as fallback if plugin does not set any.

Reply:
Ultimately, I removed this part completely. GStreamer will set us a program-like header so requests will work during actual request construction and this part should not enforce headers when not doing a web extraction.